src/main.rs

```diff
@@ -41,89 +40,66 @@ impl FromStr for PatchArg {
     }
 }
 
+const HELP_STYLES: Styles = Styles::styled()
+    .header(AnsiColor::Blue.on_default().bold())
+    .usage(AnsiColor::Blue.on_default().bold())
+    .literal(AnsiColor::White.on_default())
+    .placeholder(AnsiColor::Green.on_default());
+
 /// Test reverse dependencies with an updated crate version
-#[derive(Debug, FromArgs)]
-struct CommandLineOpts {
+#[derive(Debug, Parser)]
+#[clap(styles = HELP_STYLES)]
+struct Opts {
     /// the name of the crate to crater
-    #[argh(positional)]
     crate_name: String,
 
     /// the version of the crate to look for reverse dependencies
     /// of. Performs simple substring comparison
-    #[argh(option)]
+    #[arg(long, short)]
     version: String,
 
     /// the crate name and path to use to patch, formatted as `name=path`
-    #[argh(option)]
+    #[arg(long, short)]
     patch: Vec<PatchArg>,
 
     /// the directory to compile in.
-    #[argh(option, default = "TempDir::new().unwrap().into_path()")]
+    #[arg(long, short, default_value = TempDir::new().unwrap().path().to_string_lossy().to_string())]
     working_dir: PathBuf,
 
     /// checkout the repository for dependant instead of using a
     /// released version
-    #[argh(switch)]
+    #[arg(long, short = 'g')]
     use_git: bool,
 
     /// alternate command to run as the first step
-    #[argh(option)]
+    #[arg(long, alias = "pre")]
     pre_command: Option<String>,
 
     /// alternate command to run as the second step
-    #[argh(option)]
+    #[arg(long, alias = "post")]
     post_command: Option<String>,
-}
-
-struct Opts {
-    api_base: Url,
-    crate_name: String,
-    version: String,
-    patch: Vec<PatchArg>,
-    working_dir: PathBuf,
-    use_git: bool,
-    pre_command: Option<String>,
-    post_command: Option<String>,
-}
 
-impl Opts {
-    fn enhance(cmd: CommandLineOpts) -> Result<Self> {
-        let api_base = Url::parse(API_BASE)?;
-        let CommandLineOpts {
-            crate_name,
-            version,
-            patch,
-            working_dir,
-            use_git,
-            pre_command,
-            post_command,
-        } = cmd;
-        Ok(Self {
-            api_base,
-            crate_name,
-            version,
-            patch,
-            working_dir,
-            use_git,
-            pre_command,
-            post_command,
-        })
-    }
+    /// dry run, don't actually run the experiments
+    #[arg(long, short = 'n')]
+    dry_run: bool,
 }
 
 fn main() -> Result<()> {
-    tracing_subscriber::fmt::init();
-    let original_dir = env::current_dir()?;
-
-    let opts = Opts::enhance(argh::from_env())?;
+    color_eyre::install()?;
+    let env_filter = EnvFilter::builder()
+        .with_default_directive(LevelFilter::INFO.into())
+        .from_env_lossy();
+    tracing_subscriber::fmt().with_env_filter(env_filter).init();
```

Comment:
The ratatui changes affected logging as I now slurp it all up into a `Vec` and show it live. That could potentially now be expanded with filtering options that maybe even use the env filter format.